firmware/thruna/FSM/Src/Io/Io_AcceleratorPedals.c

```diff
@@ -1,76 +1,127 @@
+#define SAPPS_UNPRESSED_POT_V (1.18f)
+
 // Denominator term for angle calculation using the cosine law
-#define PAPPS_COS_LAW_DENOMINATOR (2.0f * PAPPS_LEN_A * PAPPS_LEN_B)
-#define PAPPS_COS_LAW_COEFFICIENT (PAPPS_LEN_A * PAPPS_LEN_A + PAPPS_LEN_B * PAPPS_LEN_B) / PAPPS_COS_LAW_DENOMINATOR
+#define PAPPS_COS_LAW_DENOMINATOR (2.0f * PAPPS_LEN_A * LEN_B) // always same because it is screwed in
+#define PAPPS_COS_LAW_COEFFICIENT ((PAPPS_LEN_A * PAPPS_LEN_A + LEN_B * LEN_B) / PAPPS_COS_LAW_DENOMINATOR)
+
+#define SAPPS_COS_LAW_DENOMINATOR (2.0f * SAPPS_LEN_A * LEN_B) // always same because it is screwed in
+#define SAPPS_COS_LAW_COEFFICIENT ((SAPPS_LEN_A * SAPPS_LEN_A + LEN_B * LEN_B) / SAPPS_COS_LAW_DENOMINATOR)
+
 // Calculate the distance of the pedal from the voltage
-#define PAPPS_LEN_PER_VOLTAGE \
-    ((PAPPS_LENGTH_FULL_PRESSED_MM - PAPPS_LENGTH_UNPRESSED_MM) / (PAPPS_FULL_PRESSED_POT_V - PAPPS_UNPRESSED_POT_V))
+// Macro to calculate the linear change in PAPPS length per change in voltage
+#define PAPPS_LEN_PER_VOLTAGE ((PAPPS_LENGTH_FULL_PRESSED_MM - PAPPS_LENGTH_UNPRESSED_MM) / (PAPPS_FULL_PRESSED_POT_V - PAPPS_UNPRESSED_POT_V))
 // PAPPS length offset in mm
-#define PAPPS_LEN_OFFSET_MM (227.6f)
+#define PAPPS_LEN_OFFSET_MM (PAPPS_LENGTH_UNPRESSED_MM - PAPPS_LEN_PER_VOLTAGE * PAPPS_UNPRESSED_POT_V) // Yint calculated based on the measured unpressed length resulting in the unpressed pot val
 #define PAPPS_RAW_VOLTAGE_TO_LEN_MM(voltage) (voltage * PAPPS_LEN_PER_VOLTAGE + PAPPS_LEN_OFFSET_MM)
 
-#define MIN_PEDAL_VOLTAGE (0.5f)
-#define MAX_PEDAL_VOLTAGE (3.1f)
+#define SAPPS_LEN_PER_VOLTAGE ((SAPPS_LENGTH_FULL_PRESSED_MM - SAPPS_LENGTH_UNPRESSED_MM) / (SAPPS_FULL_PRESSED_POT_V - SAPPS_UNPRESSED_POT_V))
+// PAPPS length offset in mm
+#define SAPPS_LEN_OFFSET_MM (SAPPS_LENGTH_UNPRESSED_MM - SAPPS_LEN_PER_VOLTAGE * SAPPS_UNPRESSED_POT_V) // Yint calculated based on the measured unpressed length resulting in the unpressed pot val
+#define SAPPS_RAW_VOLTAGE_TO_LEN_MM(voltage) (voltage * SAPPS_LEN_PER_VOLTAGE + SAPPS_LEN_OFFSET_MM)
+
+#define PAPPS_MIN_V (PAPPS_UNPRESSED_POT_V - 0.5f)
+#define PAPPS_MAX_V (PAPPS_FULL_PRESSED_POT_V + 0.5f)
+
+#define SAPPS_MIN_V (SAPPS_UNPRESSED_POT_V - 0.5f)
+#define SAPPS_MAX_V (SAPPS_FULL_PRESSED_POT_V + 0.5f)
 
-// NONLINEAR APPROXIMATION
-#define FULLY_UNPRESSED_POT_LENGTH_MM (212.2f)
-#define FULLY_PRESSED_POT_LENGTH_MM (165.6f)
+#define DEAD_ZONE_PERCENT (10.0f)
 
-#define PAPPS_DEADZONE_PERCENT (10.0f)
-#define SAPPS_DEADZONE_PERCENT (8.0f)
+
+// function created to calculate angles relative to rest angle (since we dont want a negative angle we will do rest pos - current pos)
+float Io_AcceleratorPedals_Angle_calc(float cos_law_coefficent, float cos_law_denominator, float pot_len, float max_len)
+{
+    float angle = (acosf(cos_law_coefficent - (powf(max_len, 2) / cos_law_denominator)))  -
+                  (acosf(cos_law_coefficent - (powf(pot_len, 2) / cos_law_denominator)));
```

Comment:
Hmm I would've thought that the `cos_law_denominator` should divide everything inside the `acosf` function

Reply:
it does, the way the coefficient macros works is that it performs the division on what would be a and b of the cosine law within its calculation.